### PR TITLE
Query string length for schema in characters in addition to bytes

### PR DESCRIPTION
**Pull request by partap**
Add a field for string length when querying model properties.

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

File: lib/discovery.js [hunk 1]
```diff
@@ -144,6 +144,7 @@ function mixinDiscovery(MySQL) {
       sql = paginateSQL('SELECT table_schema AS "owner",' +
           ' table_name AS "tableName", column_name AS "columnName",' +
           ' data_type AS "dataType",' +
+          ' character_maximum_length AS "length",' +
           ' character_octet_length AS "dataLength",' +
```
Comment (raymondfeng) [inline lib/discovery.js]: Can we just replace `dataLength` with `character_maximum_length`?

File: lib/discovery.js [hunk 2]
```diff
@@ -156,6 +157,7 @@ function mixinDiscovery(MySQL) {
       sql = paginateSQL('SELECT table_schema AS "owner",' +
           ' table_name AS "tableName", column_name AS "columnName",' +
           ' data_type AS "dataType",' +
+          ' character_maximum_length AS "length",' +
```
Comment (raymondfeng) [inline lib/discovery.js]: Same as above.

**Comment by raymondfeng** [comment]
@partap Any chance to make the change as I suggested?

**Comment by partap** [comment]
@raymondfeng 
Sorry, I left a note in the other issue...

I didn't want to replace `dataLength` because it could break software that may be relying on the current behavior of byte length.

**Comment by partap** [comment]
@raymondfeng 
Hmm, I checked with a few different datatypes.  It seems that with blobs it does return the byte length value, so the only time it is different is when using a multibyte character set.  

So, it looks like the only backward compatibility issue _might_ be software that takes into account the extra space allocated for multibyte strings. Your call.

**Comment by raymondfeng** [comment]
I think it's just a bug that we need to fix. We use the mysql.dataLength for the column length (in chars) during `automigrate`. The discovered result should match that.

**Comment by partap** [comment]
@raymondfeng ok.

**Pull request merged**